gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/ResourceId.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.gcloud.resourcemanager;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Represents a Google Cloud Resource Manager Resource ID
+ */
+public class ResourceId implements Serializable {
+
+  private static final long serialVersionUID = 7928469304338358885L;
+  private final String id;
+  private final Type type;
+
+  public enum Type {
+    ORGANIZATION("organization"),
+    UNKNOWN("unknown");
+
+    private final String strValue;
```

Comment:
This field is not really necessary now as we could do `ORGANIZATION.toString().toLowerCase()`. Do you prefer to keep this for future additions to admissible types?

If we expect future changes we could use a `String` for type, instead of an enum as we just did for `location` and `storageClass` in GCS.